bayesian/api/api_v2.py

```diff
@@ -172,6 +182,49 @@ def get(ecosystem, package, version):
         raise HTTPError(404, msg)
 
 
+class StackAnalysesApi(Resource):
+    """Implements stack analysis REST APIs.
+
+    Implements /api/v2/stack-analyses REST APIs for POST and GET calls.
+    """
+
+    method_decorators = [login_required]
+
+    @staticmethod
+    def get(external_request_id):
+        """Handle /api/v2/stack-analyses GET REST API."""
+        logger.debug("GET request_id: {}".format(external_request_id))
+
+        # 1. Assemble final GET response and return
+        sa_response_builder = StackAnalysesResponseBuilder(external_request_id,
+                                                           RdbAnalyses(external_request_id))
+
+        # 2. If there was no exception raise, means request is ready to be served.
+        return sa_response_builder.get_response()
+
+    @staticmethod
+    def post():
+        """Handle /api/v2/stack-analyses POST REST API."""
+        sa_post_request = None
+        try:
+            # 1. Validate and build request object.
+            sa_post_request = StackAnalysesPostRequest(**request.form, **request.files)
+
+        except ValidationError as e:
+            # 2. Check of invalid params and raise exception.
+            error_message = 'Validation error(s) in the request.'
+            for error in e.errors():
+                error_message += ' {} => {}.'.format(error['loc'][0], error['msg'])
+            logger.exception(error_message)
+            raise HTTPError(400, error=error_message)
+
+        # 3. Initiate stack analyses object
+        sa = StackAnalyses(None, sa_post_request)
```

Comment:
Probably using keyword args would avoid passing None for first param.
`StackAnalyses(params= sa_post_request)`

Reply:
I think now i don't need external_request_id parameter itself for this class now. So, now I have only one parameter as 'params' and its mandatory. So keyword args approach is not required.